src/app/refa/stages/[id]/page.tsx

```diff
@@ -1,70 +1,103 @@
-import React from 'react'
-import { Card, CardContent, CardHeader, CardTitle } from '~/components/ui/card'
-import { api } from '~/trpc/server'
+import React from "react";
+import { api } from "~/trpc/server";
 import {
   Sheet,
   SheetContent,
   SheetDescription,
   SheetHeader,
   SheetTitle,
   SheetTrigger,
-} from "~/components/ui/sheet"
-import { Badge } from '~/components/ui/badge'
-import { Separator } from '~/components/ui/separator'
+} from "~/components/ui/sheet";
+import { Badge } from "~/components/ui/badge";
+import { Separator } from "~/components/ui/separator";
+import CardWrapper from "~/components/ui/cardWrapper";
 
-
-export default async function Page({ params }: Readonly<{ params: { id: string } }>) {
-  const areas = await api.refa.areasByStage({ stageNumber: parseInt(params.id) })
+export default async function Page({
+  params,
+}: Readonly<{ params: { id: string } }>) {
+  const areasWithArtefacts = await api.refa.artifactsByStage({
+    stageNumber: parseInt(params.id),
+  });
   // if the link is not something like refa/stages/1 or refa/stages/2
   // redirect to the first stage
 
   return (
-    <div className='relative bg-white w-full min-h-full rounded-lg p-8'>
-      <div className='flex flex-wrap gap-4 min-w-full'>
-        {areas?.areas.map((area) => (
+    <div className="relative min-h-full w-full rounded-lg bg-white p-8">
+      <div className="flex min-w-full flex-wrap gap-4">
+        {areasWithArtefacts?.areas.map((area) => (
           <Sheet key={area.area_id}>
-            <SheetTrigger>
-              <Card title={area.area_name} className='p-2 m-2 min-w-fit'>
-                <CardHeader className='flex flex-col items-center justify-center' title={area.area_name}>
-                  <CardTitle className='text-center'>{area.area_name}</CardTitle>
-                </CardHeader>
-                <CardContent className='flex flex-col items-center justify-center'>
-                  <div className='text-center'>
-                    <p className='text-sm text-gray-500'>{area.area_id}</p>
-                  </div>
-                </CardContent>
-              </Card>
-            </SheetTrigger>
-            <SheetContent className='lg:min-w-[40vw]'>
+            <CardWrapper
+              area={{ id: area.area_id, name: area.area_name }}
+              artifacts={area.artefacts.map((artefact) => ({
+                id: artefact.artefact_id,
+                name: artefact.artefact_name,
+              }))}
+              // SheetTrigger is in a Area Card now.
```

Comment:
That was a little hard to navigate, but I get the approach

Reply:
A little worried about the amount of separation being done here. I personally don't mind having a lot of stuff in the same file.

Reply:
In this case, I was trying to distinguish between client and server files. As I added states to keep the chosen stage underlined